705.py

```diff
@@ -0,0 +1,50 @@
+# N.B.
+# We should go back visit linkedlist and BST solutions.
+# We should do time complexity analysis for each HashSet function.
+
+class MyHashSet:
+
+    def __init__(self):
+        """
+        Initialize your data structure here.
+        """
+        self.SPACE_SIZE = 9999
+        self.space = [[] for _ in range(self.SPACE_SIZE)]
+
+    def add(self, key: int) -> None:
+        hashcode = self.computeHash(key)
+
+        for k in self.space[hashcode]:
+            if key == k:
+                return
+
+        self.space[hashcode].append(key)
+
+    def remove(self, key: int) -> None:
+        hashcode = self.computeHash(key)
```

Comment:
Looking at the source code of `HashMap.java` in Java 8, we can tell that hash, hash code, and bucket location are three different concepts. [Line 336](https://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/java/util/HashMap.java#l336) is where the `hash` is computed based on the `key` object's `hashCode()`. [Line 569](https://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/java/util/HashMap.java#l569) shows how the `hash` is mapped to a location in the bucket, which is the `tab` variable in the source code. `tab` is a local alias of the `table` variable declared in [line 395](https://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/java/util/HashMap.java#l395).